libs/wingsdk/src/std/struct.ts

```diff
@@ -0,0 +1,32 @@
+import { T1 } from "./generics";
+import { Code, InflightClient } from "../core";
+import { Json } from "./json";
+
+/**
+ * Base class for all structs
+ * 
+ * @typeparam T1
+ */
+export class Struct {
+  /**
+   * @internal
+   */
+  public static _toInflightType(): Code {
+    return InflightClient.forType(__filename, this.name);
+  }
+
+  private constructor() { }
+
+  /**
+   * Converts a json to struct
+   * 
+   * This macro takes a json object and the struct definition file name then calls the validate method
+   * 
+   * @macro ((j, s) => {const errs = s.validate(j); if(errs.length !== 0) {throw new Error(`Unable to create struct: ${s.name} from json: ${JSON.stringify(j)}`)} return j;})($args$, $self$)
```

Comment:
Are we not adding the actual errors into the thrown exception? Is that why you `console.log` them in the validate function? If so you should make sure you `console.log` to stderr somehow. But don't you think it'll be better to add them to the exception and avoid outputing the the schema validate function?

Reply:
I think the errors should be accumulated and included in the error message, not emitted by the validation code

Reply:
Yes I remember telling myself this should probably not be a console.log at some point but I never went back to readdress 😅totally agree I’ll just return the array of errors and have them in the exception